src/scaler/simplescaler.rs

```diff
@@ -0,0 +1,387 @@
+                                    host_id,
+                                    count: remaining,
+                                }));
+                                remaining = 0;
+                            }
+                        }
+
+                        commands
+                    } else {
+                        HashSet::new()
+                    }
+                }
+                None => {
+                    if let Some(host_id) = store
+                        .list::<Host>(&self.config.lattice_id)
+                        .await?
+                        .iter()
+                        .next()
+                        .map(|(host_id, _host)| host_id)
+                    {
+                        HashSet::from_iter([Command::StartActor(StartActor {
+                            reference: self.config.actor_reference.to_owned(),
+                            count: self.config.replicas,
+                            host_id: host_id.to_owned(),
+                        })])
+                    } else {
+                        return Err(anyhow::anyhow!(
+                            "No hosts running, unable to return actor start commands"
+                        ));
+                    }
+                }
+            },
+        )
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::{collections::HashMap, sync::Arc};
+
+    use crate::{
+        commands::{Command, StartActor},
+        consumers::{manager::Worker, ScopedMessage},
+        events::{ActorClaims, ActorStarted, Event, HostStarted},
+        scaler::{simplescaler::SimpleActorScaler, Scaler},
+        test_util::TestStore,
+        workers::EventWorker,
+    };
+
+    #[tokio::test]
+    async fn can_return_error_with_no_hosts() {
+        let lattice_id = "hoohah_no_host";
+        let actor_reference = "fakecloud.azurecr.io/echo:0.3.4".to_string();
+        let replicas = 12;
+
+        let store = Arc::new(TestStore::default());
+        let simple_scaler =
+            SimpleActorScaler::new(actor_reference, lattice_id.to_string(), replicas);
+
+        let cmds = simple_scaler.reconcile(store).await;
+        assert!(cmds.is_err());
+        assert_eq!(
+            cmds.unwrap_err().to_string(),
+            "No hosts running, unable to return actor start commands".to_string()
+        );
+    }
+
+    #[tokio::test]
+    async fn can_request_start_actor() {
+        let lattice_id = "hoohah_start_actor";
+        let actor_reference = "fakecloud.azurecr.io/echo:0.3.4".to_string();
+        let replicas = 12;
+
+        // *** STATE SETUP BEGIN ***
+        // Lattice State: One empty host
+
+        let store = Arc::new(TestStore::default());
+        let worker = EventWorker::new(store.clone(), HashMap::default());
+
+        let host_id = "NASDASDIAMAREALHOST".to_string();
+        let host_name = "I am a real host".to_string();
+
+        let labels = HashMap::from([("real".to_string(), "true".to_string())]);
+
+        worker
```

Comment:
Nit: You could make your life easier if you want and just do `store.store(lattice_id, Host{})`. It is also probably better than just setting the acker to `None` 😆

Reply:
Oh, I just ended up not worrying about it. But will do this from now on (if I think about it)!